lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2128,23 +2110,31 @@ namespace Js
         wchar_t *outStr = builder.DangerousGetWritableBuffer();
 
         wchar_t* outStrLim = outStr + count;
+        wchar_t *o = outStr;
+
+        while (o < outStrLim)
+        {
+            *o++ = *inStr++;
+        }
```

Comment:
One reason could be that clang may fail to optimize the loop with pointer. Don't know VC++

As far as I remember, indexed loop would perform better (depends to the data size also) and you can compare the optimized assembly using the command line below;
```
cc -o3 -S -mllvm --x86-asm-syntax=att source.c -o output.asm
```
+1 on using `memcpy`